### PR TITLE
Improve table of contents generation.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by bgrant0607** [comment]
LGTM

**Comment by bgrant0607** [comment]
cc @lavalamp

**Comment by lavalamp** [comment]
lgtm

**Comment by lavalamp** [comment]
I need to fix this in the link checker, too.

**Comment by bgrant0607** [comment]
cc @thockin for merge

File: docs/accessing-the-cluster.md [hunk 1]
```diff
@@ -1,9 +1,27 @@
 # User Guide to Accessing the Cluster
- * [Accessing the cluster API](#api)
- * [Accessing services running on the cluster](#otherservices)
- * [So many proxies](#somanyproxies)
 
-## Accessing the cluster API<a name="api"></a>
+**Table of Contents**
+<!-- BEGIN GENERATED TOC -->
```
Comment (thockin) [inline docs/accessing-the-cluster.md]: This needs to become the new universal munge tag:

Comment (thockin) [inline docs/accessing-the-cluster.md]: Same for all files

**Comment by erictune** [comment]
fixes #11040

**Comment by erictune** [comment]
rebased on @thockin changes.

**Comment by erictune** [comment]
@lavalamp you wanted this to go in so feel free to review.

**Comment by lavalamp** [comment]
LGTM

**Comment by erictune** [comment]
@thockin ok to merge?

**Comment by thockin** [comment]
LGTM

**Comment by bgrant0607** [comment]
Shippable failure:
```
PATH=$GOPATH/bin:./third_party/etcd:$PATH ./hack/verify-gendocs.sh
docs/user-guide/connecting-applications.md
----
table-of-contents:
contents were modified
docs/volumes.md
----
table-of-contents:
contents were modified
FAIL: changes needed but not made due to --verify
!!! Error in ./hack/verify-gendocs.sh:51
'"${mungedocs}" "--verify=true" "--root-dir=${DOCROOT}"' exited with status 1
Call stack:
1: ./hack/verify-gendocs.sh:51 main(...)
Exiting with status 1
```

**Comment by erictune** [comment]
I can't fix this until #11187

**Comment by thockin** [comment]
@k8s-bot retest this please

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by thockin** [comment]
1/2 shippable OK, other is a timeout.  Going.

**Pull request merged**